src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -9,6 +9,45 @@
 
     @Override
     public User register(User user) {
+        if(userValidation(user)){
+            return storageDao.add(user);
+        }
         return null;
     }
+
+
+    private boolean userValidation(User user) {
+        return userNullCheck(user)
+                && userValidateLoginExist(user.getLogin())
+                && userValidateLoginLength(user.getLogin())
+                && userValidatePasswordLength(user.getPassword())
+                && userValidateAge(user.getAge());
+    }
+
+    private boolean userNullCheck(User user){
+        return user != null
+                && user.getLogin() != null
+                && user.getPassword() != null
+                && user.getAge() != null;
+    }
+
+    private boolean userValidateLoginExist(String userLogin) {
+        return storageDao.get(userLogin) == null;
+    }
+
+    private boolean userValidateLoginLength(String userLogin) {
+        return userLogin.length() >= 6;
+    }
+
+    private boolean userValidatePasswordLength(String userPassword) {
+        return userPassword.length() >= 6;
+    }
+
+    private boolean userValidateAge(int age) {
+        return age >= 18;
+    }
```

Comment:
The `userValidateAge` method checks if the age is at least 18. Ensure that the `User` class has an `Integer` type for age to match the null check in `userNullCheck`.